### DISCUSSION TITLE
Pane select customization

### DISCUSSION THREAD

**Discussion by nikola993** [wezterm/wezterm#4477] [General]
Is there a way after I select a pane with pane select action for that pane to be zoomed in? I looked for option and did not find anything.

I tried creating something like that in config but my lua knowlage is really basic. Can that be implemented in the pane select action as an option or is there any good soul who can write some example of the config and how is that possible to do in lua.

**Comment by bew** [comment]
Ideally there would be a `PaneSelectArgs` action that allows to do an arbitrary action of the selected pane, until then you could try to use a combination of:
- [`PaneSelect`](https://wezfurlong.org/wezterm/config/lua/keyassignment/PaneSelect.html)
- [`SetPaneZoomState`](https://wezfurlong.org/wezterm/config/lua/keyassignment/SetPaneZoomState.html)

The only thing though is I don't know if using both one after the other will actually wait for the pane to be selected to do the 2nd action.. (which is necessary here..)

**Reply by nikola993** [reply]
I tried that already, I tried using action.Multiple
Adding in keys config something like this is not working.

    action = act.Multiple {
      act.SetPaneZoomState(false),
      act.PaneSelect,
      act.SetPaneZoomState(true),
    },

After entering the pane select mode SetPaneZoomState is triggered before the selection of the pane is finished. I tried to find event like on_pane_change to trigger second pane zoom state change but didn't worked.

**Reply by bew** [reply]
Maybe you can update wezterm, as the behavior you're describing is already the default behavior of `PaneSelect` since #3573 , released as part of version [20230712-072601-f4abf8fd](https://wezfurlong.org/wezterm/changelog.html#20230712-072601-f4abf8fd) (4th bullet point)

**Reply by nikola993** [reply]
Yeah it is working as expected with the update, I had updated the wezterm but apparently I never restarted the app for updates to take effect. Thanks for the help!
